src/java/org/apache/cassandra/io/sstable/format/SSTableWriter.java

```diff
@@ -439,7 +441,8 @@ public static void hardlink(Descriptor tmpdesc, Descriptor newdesc, Set<Componen
 
     public static void guardCollectionSize(TableMetadata metadata, DecoratedKey partitionKey, Unfiltered unfiltered)
```

Comment:
I noticed this method is called only from the same class. It should be private. It should be private because it would be actually a bug to call it with a different metadata than the one for this table.

Reply:
<img width="638" alt="Screenshot 2022-08-09 at 17 33 38" src="https://user-images.githubusercontent.com/102972224/183709206-810eec40-6c91-42b4-b72f-85568bb5c185.png">

I tried it with private but then I noticed an issue being flagged in SortedTableWriter.java class. As suggested, I tried making it package-private which gives private access to the same package and resolves the error, but not sure if it will cause further issues to the rest of the project. If you could recommend me something about it.

Reply:
Package private is ok. A bigger problem is the static part. There are at least two ways I can see:
- move your optimization one level up, so don't touch guardCollectionSize at all,
- refactor this so guardCollectionSize is non-static, private and possibly duplicated between those two classes. Then extract common *static* code of it to something else.